packages/i18next/src/useCurrentLanguage.ts

```diff
@@ -0,0 +1,7 @@
+import { useI18nextPlugin } from "./useI18nextPlugin.ts";
+
+export function useCurrentLanguage() {
+    const plugin = useI18nextPlugin();
+
+    return plugin.currentLanguage;
```

Comment:
plugin.currentLanguage is probably typed with a string, instead of the expected language types

Reply:
Yes, IMO it the responsability of the consumer to cast it to it's specific language type if he want's something more restrictive, like:

```
const language = useCurrentLanguage as LanguageKey;
```

Reply:
I will add a documentation example typing the return of `useCurrentLanguage()` in the consumer code to make sure devs are aware of this.

Reply:
As a user, i honestly feel like i wouldn't use these method. First thing i would do is 

in sample/shared
```
import { useI18nextPlugin as useI18nextPluginFireFly } from "..firefly"

function useI18nextPlugin() {
    return useI18nextPluginFireFly() as i18Next<LanguageKey>;
}

function useCurrentLanguage() { 
    return useI18nextPlugin().language
}

function useChangeLanguage() { 
    const plugin = useI18nextPlugin();

    return useCallback<typeof plugin.changeLanguage>(language => {
        return plugin.changeLanguage(language);
    }, [plugin]);
}


```

Reply:
I do expect user to wrap these hooks if they prefer to offer a better typing experience.

I don't think that offering generic parameters in these cases would be appropriate because that would go against our previous stance of prefering that type of casting to be done in consumers land and would lead to an inconstent API.

I do believe thought that it would preferable for consumers to wrap the hooks rather than the plugin instance itself because the hooks could introduce optimizations that they wouldn't know about:

```
import { useCurrentLanguage as useSquideCurrentLanguage } from `@squide/i18next`

export function useCurrentLanguage() { 
    return useSquideCurrentLanguage() as LanguageKeu
}
```

Reply:
Should the sample use this pattern to showcase it?

Reply:
I prefer the samples to be as close as possible to the reference documentation. I do showcase the most important patterns. While this one significative, I don't feel like it's one of the most important one.